pkg/languagedetection/detector.go

```diff
@@ -0,0 +1,46 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package languagedetection
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+)
+
+type Language struct {
+	Name string
+}
```

Comment:
You might want to use some kind of enum to define valid languages if you have a predefined list, it would probably be easier to handle than an arbitrary string

Reply:
I think that makes sense. I do want these to be strings so they're embeddable in tags, but I can turn this into an enum.